CompetitionTemplate/src/competition/subsystems/vision/JetsonCommPacket.java

```diff
@@ -0,0 +1,190 @@
+     */
+
+    /**
+     * Processes a newly-received integer value and updates internal state tracking accordingly.
+     * @param newValue the new value to process.
+     * @return A boolean indicating whether more data are needed to fill this packet.
+     */
+    public boolean addNewValue(int newValue) {
+        switch (currentParserState) {
+            case WAITING_FOR_START:
+                // Skip data until we find the start of a packet
+                if (newValue == startOfPacketInt) {
+                    this.currentParserState = PacketParserState.WAITING_FOR_PAYLOAD_TYPE_FLAG;
+                }
+                break;
+
+            case WAITING_FOR_PAYLOAD_TYPE_FLAG:
+                this.payloadType = PacketPayloadType.parse(newValue);
+                
+                if(this.payloadType == null) {
+                    currentParserState = PacketParserState.MALFORMED_PACKET_ABORT;
+                    // If we get junky data, give up
+                    return false;
+                }
+                else {
+                    currentParserState = PacketParserState.WAITING_FOR_PAYLOAD_LENGTH;
+                }
+                break;
+
+            case WAITING_FOR_PAYLOAD_LENGTH:
+                this.expectedPayloadLength = newValue;
+                
+                if(this.expectedPayloadLength > 0) {
+                    currentParserState = PacketParserState.WAITING_FOR_PAYLOAD_DATA;
+                }
+                else if (this.expectedPayloadLength == 0) {
+                    currentParserState = PacketParserState.PARSE_COMPLETE;
+                    
+                    return false;
+                }
+                else {
+                    currentParserState = PacketParserState.MALFORMED_PACKET_ABORT;
+                    // If we get junky data, give up
+                    return false;
+                }
+                break;
+
+            case WAITING_FOR_PAYLOAD_DATA:
+                this.packetPayloadData.add((Integer) newValue);
+                
+                if (packetPayloadData.size() >= expectedPayloadLength) {
+                    currentParserState = PacketParserState.PARSE_COMPLETE;
+                    return false;
+                }
+                break;
+            case PARSE_COMPLETE:
+                // We shouldn't get here; consumer should stop after all data is received
+                return false;
+            case MALFORMED_PACKET_ABORT:
+                return false;
+            default:
+                // Do nothing
+        }
+
+        return true;
+    }
+    
+    public boolean addNewValues(int[] newValues) {
+        if(newValues == null || newValues.length <= 0) {
+            log.error("No values given to add!");
+            return !isParseComplete();
+        }
+        
+        boolean shouldContinue = true;
+        for(int i = 0;
+                i < newValues.length && (shouldContinue = addNewValue(newValues[i]));
```

Comment:
I feel like this should continue side effect is a little unusual for Java, would maybe vote to pull it out to be explicitly set in the loop